index.js

```diff
@@ -149,6 +150,90 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Adds the Screwdriver webhook to the Github repository
+     * @method _addWebhook
+     * @param  {Object}    config            Config object
+     * @param  {String}    config.scmUri     The SCM URI to add the webhook to
+     * @param  {String}    config.token      Service token to authenticate with Github
+     * @param  {String}    config.webhookUrl The URL to use for the webhook notifications
+     * @return {Promise}                     Resolve means operation completed without failure.
+     */
+    _addWebhook(config) {
+        const searchForHook = (scmInfo, url, page) => {
+            const getHooks = this.breaker.runCommand({
+                action: 'getHooks',
+                token: config.token,
+                params: {
+                    owner: scmInfo.owner,
+                    repo: scmInfo.repo,
+                    page,
+                    per_page: WEBHOOK_PAGE_SIZE
+                }
+            });
+            const filterForHook = getHooks.then((hooks) => {
+                let found = null;
+
+                hooks.some((hook) => {
+                    const currentHook = hoek.reach(hook, 'config.url');
+
+                    if (currentHook === url) {
+                        found = hook;
+                    }
+
+                    return found;
+                });
+
+                return found;
+            });
+
+            return Promise.all([
+                getHooks,
+                filterForHook
```

Comment:
This is a "perfectly cromulent" way to use Promises, but from a reader's perspective it does look pretty funky. Were `findInHook` to return `{ hooks, screwdriverHook: found }` then all the weirdness of the `Promise.all` goes away since you resolve the chain a little cleaner: `findInHook.then(({ hooks, screwdriverHook }) => { ... })`;

Overall, for something that is looking to reduce complexity, `_addWebhook` has a lot of code in it.